packages/tre/src/plugins/shared/routing.ts

```diff
@@ -0,0 +1,142 @@
+import { URL, domainToUnicode } from "url";
+import { MiniflareError } from "../../shared";
+
+export type RouterErrorCode = "ERR_QUERY_STRING" | "ERR_INFIX_WILDCARD";
+
+export class RouterError extends MiniflareError<RouterErrorCode> {}
+
+export interface WorkerRoute {
+  target: string;
+  route: string;
+
+  protocol?: string;
+  allowHostnamePrefix: boolean;
+  hostname: string;
+  path: string;
+  allowPathSuffix: boolean;
+}
+
+const A_MORE_SPECIFIC = -1;
+const B_MORE_SPECIFIC = 1;
+
+export function parseRoutes(allRoutes: Map<string, string[]>): WorkerRoute[] {
```

Comment:
I imagine there's code to do this in EWC somewhere—have you checked the logic against that? (I'm assuming this is copied from Miniflare 2)